src/app/devices/sensor-details/sensor-detail.component.test.ts

```diff
@@ -126,4 +127,43 @@ describe('sensor-detail', () => {
         sut.cancel();
         expect(router.navigate).toHaveBeenCalledWith(['/devices']);
     });
+
+    it('should not allow both executor and servo to be checked at one time (servo changed)', (done) => {
+        const sensorMock = new Sensor({
+            servo: true,
+            executor: true
+        });
+
+        sut.sensor = sensorMock;
+        sut.needUpdate = true;
+        sut.validate('servo');
+        expect(sut.sensor.executor).toBe(false);
+        done();
+    });
+
+    it('should not allow both executor and servo to be checked at one time (executor changed)', (done) => {
+        const sensorMock = new Sensor({
+            servo: true,
+            executor: true
+        });
+
+        sut.sensor = sensorMock;
+        sut.needUpdate = true;
+        sut.validate('executor');
+        expect(sut.sensor.servo).toBe(false);
+        done();
```

Comment:
There is no async stuff to call **done**, you can totally remove done from arguments